pkg/operator/controller/ingress/status.go

```diff
@@ -204,18 +203,17 @@ func computeAllowedSourceRanges(service *corev1.Service) []operatorv1.CIDR {
 	return nil
 }
 
-// computeDeploymentPodsScheduledCondition computes the ingress controller's
-// current PodsScheduled status condition state by inspecting the PodScheduled
-// conditions of the pods associated with the deployment.
-func computeDeploymentPodsScheduledCondition(deployment *appsv1.Deployment, pods []corev1.Pod) operatorv1.OperatorCondition {
+// checkPodsScheduledForDeployment checks whether the given deployment has a
+// valid label selector and whether all of the deployment's pods are reporting
+// PodsScheduled=True.  This function returns an error value indicating the
+// result of that check.
+func checkPodsScheduledForDeployment(deployment *appsv1.Deployment, pods []corev1.Pod) error {
+	if deployment == nil {
+		return errors.New("Deployment was nil.  This is a bug!")
```

Comment:
```suggestion
		return errors.New("System or configuration error detected - deployment was nil.")
```

Reply:
Then maybe just "System error detected." and put a link to the bug reporting tool!

Reply:
I updated the error message as you suggested.  The Jira link is a bit unwieldy though.